batch_writer.go

```diff
@@ -30,10 +30,21 @@ func (w *BatchWriter) Initialize() {
 
 func (w *BatchWriter) WriteRowBatch(batch *RowBatch) error {
 	return WithRetries(w.WriteRetries, 0, w.logger, "write batch to target", func() error {
-		if batch.Size() == 0 {
+		values := batch.Values()
+		if len(values) == 0 {
```

Comment:
Can we delete `Size()` then?

Reply:
It's still used in two locations in the `Cursor`. I don't want to change that for now to keep this change small.